pkg/registry/core/service/storage/rest_test.go

```diff
@@ -1251,6 +1251,7 @@ func TestServiceRegistryUpdateDryRun(t *testing.T) {
 			Selector:        map[string]string{"bar": "baz"},
 			SessionAffinity: api.ServiceAffinityNone,
 			Type:            api.ServiceTypeNodePort,
+			ClusterIP:       "1.2.3.5",
```

Comment:
This service was created at the beginning of the test but never deleted, so it could take the IP `1.2.3.4` and never release it, causing the test to fail

Reply:
I'm not sure I understand this... wasn't the service created at the beginning an externalname service? I didn't think those got an assigned clusterIP

Reply:
This one (L1248) is creating a new NodePort service, that's why I'm hardcoding the ClusterIP, so it does not get one assigned randomly that may collision

Reply:
oh, I see... *this* is the problematic service

Reply:
ic the problem, my comment is misleading, 

> This service was created at the beginning of the test but never deleted,

replace "beginning of the test" by "before the service with the 1.2.3.4 ClusterIP" :upside_down_face:

Reply:
does setting an explicit IP here remove the need to broaden the range in `makeIPNet`?

Reply:
yep, I also tested, 
https://github.com/kubernetes/kubernetes/pull/93906#issuecomment-672318048

do you prefer to keep the subnet size?

Reply:
yeah, let's keep the subnet size small (as-is) so any flakes due to non-deterministic clashing in the tests actually is observable/reproducible

Reply:
done